src/charm.py

```diff
@@ -173,7 +198,11 @@ def _on_get_certificate_action(self, event: ActionEvent) -> None:
     def _on_certificates_relation_created(self, _: RelationCreatedEvent) -> None:
         """Handle the TLS Certificate relation created event."""
         tls_information = TLSInformation.from_charm(self)
-        config = CharmConfig.from_charm(self)
+
+        field_manager = self.app.name
+        namespace = self.model.name
+        client = get_client(field_manager, namespace)
```

Comment:
I think I am not seeing something obvious, but why not instantiate this variable as a class variable in the `__init__` method?

Reply:
Maybe then another option is to have a `client` property in the Charm (and cached_property if that is necessary)

Reply:
That makes sense as well, I'll create a follow-up PR and let's move this discussion there.